### PR TITLE
Redmine 11812: Advanced export, add translation in CSV export, remove…

### PR DESCRIPTION
**Pull request by man-axelor**
… duplicate code in advancedExportController by creating a method advancedExportFile, improve code in AdvancedExportService

### CODE REVIEW

File: axelor-base/src/main/java/com/axelor/apps/base/service/AdvancedExportServiceImpl.java [hunk 1]
```diff
@@ -690,12 +686,8 @@ public MetaFile advancedExportExcel(MetaFile exportFile, MetaModel metaModel, Li
 			Cell headerCell = headerRow.createCell(colHeaderNum++);
 			
 			MetaTranslation metaTranslation = metaTranslationRepo.all().filter("self.key = ?1 and self.language = ?2",advancedExportLine.getTitle(), language).fetchOne();
```
Comment (gdu-axelor) [inline axelor-base/src/main/java/com/axelor/apps/base/service/AdvancedExportServiceImpl.java]: Why don't use I18n.get("..") method ?

File: axelor-base/src/main/java/com/axelor/apps/base/service/AdvancedExportServiceImpl.java [hunk 2]
```diff
@@ -755,41 +747,50 @@ public MetaFile advancedExportCSV(MetaFile exportFile, MetaModel metaModel, List
...
+                    Object value = field.get(colName);
+                    if (value == null || value == "") {
+                        totalCols[i++] = null;
+                    } else {
+                        totalCols[i++] = value.toString();
+                    }
+                }
+                csvWriter.writeNext(totalCols);
+                i = 0;
+            }
```
Comment (gdu-axelor) [inline axelor-base/src/main/java/com/axelor/apps/base/service/AdvancedExportServiceImpl.java]: You have to close the csvWriter or call CsvTool class.

File: axelor-base/src/main/java/com/axelor/apps/base/web/AdvancedExportController.java [hunk 3]
```diff
@@ -188,79 +191,78 @@ public void showAdvancedExportData(ActionRequest request, ActionResponse respons
...
+            advancedExportFile(request, response, "EXCEL");
+        } catch (DocumentException | ClassNotFoundException | IOException | AxelorException e) {
+            TraceBackService.trace(e);
+        }
 	}
-	
+
+	public void advancedExportCSV(ActionRequest request, ActionResponse response) {
+        try {
+            advancedExportFile(request, response, "CSV");
```
Comment (gdu-axelor) [inline axelor-base/src/main/java/com/axelor/apps/base/web/AdvancedExportController.java]: Better to use static class variable

File: axelor-base/src/main/java/com/axelor/apps/base/service/AdvancedExportServiceImpl.java (hunk 2 above)
Comment (man-axelor) [reply axelor-base/src/main/java/com/axelor/apps/base/service/AdvancedExportServiceImpl.java]: A try-with-ressource is used, with it, we don't need to close csvWriter as it close it automatically.
Try-with-ressource was introduced in Java 7.
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

**Pull request merged**